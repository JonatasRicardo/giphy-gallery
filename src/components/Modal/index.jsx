import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useTheme } from "react-jss";
import useStyles from "./modal.styles";
import Spinner from "../Spinner";

const propTypes = {
  open: PropTypes.bool,
  children: PropTypes.node,
  onClose: PropTypes.func
};

const GalleryModal = ({
    open,
    children,
    onClose = () => {}
}) => {
    const [show, setShow] = useState(open);
    const theme = useTheme();
    const classes = useStyles({ theme });

    useEffect(() => {
        setShow(open);
    }, [open, children]);

    const closeModal = () => {
        setShow(false);
        onClose();
    };

    return (
        <div className={`${classes.modal} ${['', 'is-open'][Number(show)]}`} role="dialog" data-testid="modal">
            <div className={classes.modalBody} >
                <button
                  className={classes.close}
                  onClick={closeModal}
                  data-testid="button-close-modal"
                >
                  Close
                </button>
                <div className={classes.spinner}><Spinner /></div>
                <div className={classes.content}>{children}</div>
            </div>
            <div className={classes.modalOverlay} onClick={closeModal} data-testid="overlay-modal" />
        </div>
    );
}

GalleryModal.propTypes = propTypes;

export default GalleryModal;
