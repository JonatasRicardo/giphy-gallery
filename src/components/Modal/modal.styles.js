import {
  createUseStyles
} from "react-jss";

const useStyles = createUseStyles({
  modal: {
    position: 'fixed',
    width: '100%',
    height: '100%',
    left: 0,
    top: 0,
    zIndex: -1,
    opacity: 0,
    display: 'flex',
    flexFlow: 'column',
    alignItems: 'center',
    justifyContent: 'space-around',
    transition: 'all .5s',
    '&.is-open': {
      animation: '$fadeIn .15s forwards'
    }
  },
  close: {
    background: ({
      theme
    }) => theme.dark.background,
    color: ({
      theme
    }) => theme.dark.primary,
    border: 0,
    padding: ({
      theme
    }) => theme.spacing,
    outline: 'none',
    cursor: 'pointer'
  },
  spinner: {
    position: 'absolute',
    left: '50%',
    top: '50%',
    margin: '-20px 0 0 -20px'
  },
  content: {
    position: 'relative'
  },
  modalBody: {
    zIndex: 1,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-end',
    padding: ({
      theme
    }) => theme.spacing
  },
  modalOverlay: {
    background: ({
      theme
    }) => theme.dark.primary,
    opacity: .6,
    position: 'absolute',
    width: '100%',
    height: '100%'
  },
  '@keyframes fadeIn': {
    from: {
      zIndex: -1,
      opacity: 0
    },
    to: {
      zIndex: 1,
      opacity: 1
    }
  },
  '@keyframes fadeOut': {
    from: {
      zIndex: 1,
      opacity: 1

    },
    to: {
      zIndex: -1,
      opacity: 0
    }
  }
});

export default useStyles;
