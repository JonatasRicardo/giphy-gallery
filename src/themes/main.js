const theme = {
  dark: {
    background: "#000",
    primary: "#fff"
  },
  light: {
    background: "#fff",
    primary: "#000"
  },
  spacing: 10
};

export default theme;
