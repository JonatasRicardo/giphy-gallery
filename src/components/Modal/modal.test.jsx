import React from 'react';
import { ThemeProvider } from "react-jss";
import {
  render
} from '@testing-library/react';
import Modal from './index';
import themes from '../../themes/main';


const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <Modal />
    </ThemeProvider>
  );

test('should match snapshot', async () => {
  const { asFragment } = renderComponent({ theme: themes });
  expect(asFragment()).toMatchSnapshot()
});
