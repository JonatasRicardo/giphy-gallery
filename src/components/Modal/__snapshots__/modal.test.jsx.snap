// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal-0-2-1 undefined"
    data-testid="modal"
    role="dialog"
  >
    <div
      class="modalBody-0-2-5 modalBody-d1-0-2-10"
    >
      <button
        class="close-0-2-2 close-d0-0-2-9"
        data-testid="button-close-modal"
      >
        Close
      </button>
      <div
        class="spinner-0-2-3"
      >
        <div
          class="spinner-0-2-12"
        >
          <div
            class="doubleBounce-0-2-13 doubleBounce-d0-0-2-16"
          />
          <div
            class="doubleBounce-0-2-13 doubleBounce-d0-0-2-16,doubleBounce2-0-2-14"
          />
        </div>
      </div>
      <div
        class="content-0-2-4"
      />
    </div>
    <div
      class="modalOverlay-0-2-6 modalOverlay-d2-0-2-11"
      data-testid="overlay-modal"
    />
  </div>
</DocumentFragment>
`;
